### PR TITLE
test_migrate: ensure migrated config file is valid

### PR DESCRIPTION
**Pull request by tych0**
The result of qtile migrate should be a valid python file which should
parse and type check correctly. If mypy is present (which it is in CI),
let's check this.

Signed-off-by: Tycho Andersen <tycho@tycho.pizza>

### CODE REVIEW

**Comment by ep12** [comment]
That force-push you did to make the checks pass wasn't a clever idea I think. `tox.ini` tells me that `mypy` is only available in the `mypy` testenv and there we are running only `pytest test/test_check.py`. So this does not enable `mypy` checking in the CI (which is what you're aiming for).

**Comment by tych0** [comment]
> That force-push you did to make the checks pass wasn't a clever idea I think.

Well it fixed a bug (testing the function for truthiness vs. the function's result), but,

> tox.ini tells me that mypy is only available in the mypy testenv and there we are running only pytest test/test_check.py. So this does not enable mypy checking in the CI (which is what you're aiming for).

Yeah, right you are. I guess maybe we need to run the migrate checks in the mypy environment as well.

File: test/test_migrate.py [hunk 1]
```diff
@@ -46,8 +47,9 @@ def check_migrate(orig, expected):
             f.write(orig.encode('utf-8'))
 
         run_qtile_migrate(config_path)
+        if have_mypy():
+            run_qtile_check(config_path)
```
Comment (Fabien-Bernard) [inline test/test_migrate.py:51]: Doesn't seem to stop me doing my bad thing. Shouldn't we assert it?

Comment (tych0) [reply test/test_migrate.py:51]: Yes, looks like we should. Also looks like the mypy env doesn't have bowler... bleh.

Comment (ep12) [reply test/test_migrate.py:51]: Should we assert it? Wouldn't it be enough to run this check in one environment? I mean, we would be using the same bowler version in different test envs, that shouldn't make a difference and just makes the test runs slower...

Comment (tych0) [reply test/test_migrate.py:51]: In theory environments shouldn't make a difference at all; in practice they often do :). Yes, we should put an assert here, though.

**Comment by m-col** [comment]
Nice, LGTM.

**Pull request merged**